src/main/java/lotto/domain/LottoRank.java

```diff
@@ -17,7 +18,15 @@ public enum LottoRank {
         this.prizeMoney = prizeMoney;
     }
 
-    public static LottoRank of(int count) {
+    public static LottoRank valueOf(int count, boolean matchBonus) {
+        if (count == 5 && matchBonus) {
+            return SECOND;
+        }
+
+        if (count == 5 && !matchBonus) {
```

Comment:
위에서 보너스 볼 일치 여부에 대한 검증이 끝났으니, 여기서 추가로 검증할 필요는 없을 것 같네요.
```suggestion
        if (count == 5) {
```